src/config-file-schema.ts

```diff
@@ -29,8 +27,7 @@ function isErrnoException(error: any): error is NodeJS.ErrnoException {
 
 export function ConfigFileSchema<T>(opts: {
   path: string;
-  schema: JSONSchemaType<T>;
-  validate?: ValidateFunction;
+  validate: ValidateFunction;
```

Comment:
I wonder if this should be named `validateFunction` since as an input parameter it looks a bit like it would be a boolean